.github/workflows/test.yaml

```diff
@@ -24,22 +28,44 @@ defaults:
   run:
     shell: bash
 
+env:
+  # Set to 'yes' to open a tunnel to GitHub's VMs through tmate on failures.
+  # Also increase timeout-minutes for the relevant OS when debugging remotely
+  # and you need a longer session.
+  TMATE_DEBUG: 'no'
 
 jobs:
   testing:
-    runs-on: ubuntu-20.04
+    name: ${{ matrix.python-version }}-${{ matrix.job-name || 'lnx-default-tests' }}
+    # We have Ubuntu as the base for running agains multiple Python versions.
+    runs-on: "${{ matrix.runs-on || 'ubuntu-20.04'}}"
```

Comment:
the 'ubuntu-20.04' is not explicitly defined in the matrix so that we don't have to repeat it each time we extend it.

We leave it empty and use a default value.

Another option is to define the default values via global environment variables.

```
env:
   DEFAULT_RUNS_ON: 'ubuntu-22.04'
jobs:
  testing:
    runs-on: "${{ matrix.runs-on || 'env.DEFAULT_RUNS_ON'}}"
```